OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/services/ServiceProvider.kt

```diff
@@ -80,6 +85,26 @@ class ServiceProvider(
         }
     }
 
+    fun scheduleStartServices() {
+        synchronized(serviceMap) {
+            for (serviceReg in serviceMap!![IStartableService::class.java]!!) {
+                // we can directly start services that were previously resolved
+                if (serviceReg.isResolved()) {
+                    val service = serviceReg.resolve(this) as IStartableService
+                    service.start()
```

Comment:
We should always run `start()` off the main thread if that is the optimization we are trying to do for all of them. Is there any reason why you wouldn't?

Reply:
Good point, I just pushed a commit that stop checking isResolved and schedule start off the main thread for all services. It is much cleaner now. I have tested that this approach works the same.